docker/build.sh

```diff
@@ -9,24 +9,132 @@
 # Jenkins build job should run with all options, for example,
 #   ./docker/build.sh jar -d push
 
-set -ex
-PROJECT=leonardo
+HELP_TEXT="$(cat <<EOF
+
+ Build the Leonardo code and docker images.
+   jar : build Leonardo jar
+   -d | --docker : (default: no action) provide either "build" or "push" to
+           build or push a docker image.  "push" will also perform build.
+   -r | --registry: (default: dockerhub) can be either "dockerhub" or "gcr".
+           Users of gcr should have the gcloud tool installed and configured.
+   -p | --project: set the project used at either dockerhub or with gcr
+           container registries.
+   -h | --help: print help text.
+
+ Examples:
+   Jenkins build job should run with all options, for example,
+     ./docker/build.sh jar -d push
+   To build the jar, the image, and push it to a gcr repository.
+     ./docker/build.sh jar -d build -r gcr --project "my-awesome-project"
+\t
+EOF
+)"
+
+# Enable strict evaluation semantics.
+set -e
+
+# Set default variables used while parsing command line options.
+TARGET="${TARGET:-leonardo}"
+GIT_BRANCH="${GIT_BRANCH:-$(git rev-parse --abbrev-ref HEAD)}"
```

Comment:
Our Jenkins infrastructure sets `$BRANCH` not `$GIT_BRANCH`, so I needed to change this to the following:
```
GIT_BRANCH="${BRANCH:-$(git rev-parse --abbrev-ref HEAD)}"
```